ra/registration-authority.go

```diff
@@ -62,7 +62,7 @@ func validateEmail(address string) (err error) {
 	var mx []*net.MX
 	mx, err = net.LookupMX(domain)
 	if err != nil {
-		err = core.InternalServerError(err.Error())
+		err = core.MalformedRequestError(fmt.Sprintf("Supplied email domain %s could not be resolved.", domain))
```

Comment:
*Side note* `net.LookupMX` can return underlying transport errors not caused by the user, it can be checked for type `*net.DNSError` to see the internal error (no host/timeout/server issues/etc) but this probably isn't 100% necessary.